src/sagemaker/lineage/query.py

```diff
@@ -226,29 +226,119 @@ def __init__(
 
     def __str__(self):
         """Define string representation of ``LineageQueryResult``.
-        
+
         Format:
         {
             'edges':[
                 {
-                    'source_arn': 'string', 'destination_arn': 'string', 
+                    'source_arn': 'string', 'destination_arn': 'string',
                     'association_type': 'string'
                 },
                 ...
             ]
             'vertices':[
                 {
-                    'arn': 'string', 'lineage_entity': 'string', 
-                    'lineage_source': 'string', 
+                    'arn': 'string', 'lineage_entity': 'string',
+                    'lineage_source': 'string',
                     '_session': <sagemaker.session.Session object>
                 },
                 ...
             ]
         }
-        
+
         """
         result_dict = vars(self)
-        return (str({k: [vars(val) for val in v] for k, v in result_dict.items()}))
+        return str({k: [vars(val) for val in v] for k, v in result_dict.items()})
+
+    def _import_visual_modules(self):
+        """Import modules needed for visualization."""
+        import dash_cytoscape as cyto
+
+        from jupyter_dash import JupyterDash
+
+        from dash import html
+
+        return cyto, JupyterDash, html
+
+    def _get_verts(self):
+        """Convert vertices to tuple format for visualizer."""
+        verts = []
+        for vert in self.vertices:
+            verts.append((vert.arn, vert.lineage_source))
+        return verts
+
+    def _get_edges(self):
+        """Convert edges to tuple format for visualizer."""
+        edges = []
+        for edge in self.edges:
+            edges.append((edge.source_arn, edge.destination_arn, edge.association_type))
+        return edges
+
+    def visualize(self):
+        """Visualize lineage query result."""
+        cyto, JupyterDash, html = self._import_visual_modules()
+
+        cyto.load_extra_layouts()  # load "klay" layout (hierarchical layout) from extra layouts
+        app = JupyterDash(__name__)
+
+        verts = self._get_verts()
+        edges = self._get_edges()
+
+        nodes = [
+            {
+                "data": {"id": id, "label": label},
+            }
+            for id, label in verts
+        ]
+
+        edges = [
+            {"data": {"source": source, "target": target, "label": label}}
+            for source, target, label in edges
+        ]
+
+        elements = nodes + edges
+
+        app.layout = html.Div(
+            [
+                cyto.Cytoscape(
```

Comment:
This might be a good candidate to move into a separate class/function entirely. It has no dependencies on the member variables for LineageQueryResult, so the function call could just look like `get_cytoscape(elements)`. This would make it more testable, because we could directly call it with any elements we wanted. In the future if we wanted to allow say users to set their colors this would help keep the code cleaner too

Reply:
Added DashVisualizer class and moved dash code to this class.